Left base folder: /home/mr_halfword/Downloads/gsc_api.linux.1.6.10.2.tar.gz
Right base folder: /home/mr_halfword/GscApi-1.6.10.2
diff -r gsc_api/PlxSdk/Linux/Driver/Source.Plx9000/ApiFunc.c gsc_api/PlxSdk/Linux/Driver/Source.Plx9000/ApiFunc.c
37a38,39
> /* The following block of compatibility code was taken from https://stackoverflow.com/a/65682486/4207678
>    "How to replace the recently abolished first parameter of macro "access_ok"?
38a41,54
>    And removed the 1st parameters of calls to access_ok
> */
> 
> #include <linux/version.h>
> 
> /* 
>  * <linux/uaccess.h> was added in kernel version 2.6.18, and should be
>  * included in preference to <asm/uaccess.h>.  In particular, copy_to_user()
>  * and copy_from_user() were moved to <linux/uaccess.h> in kernel version
>  * 4.12.
>  */
> #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)
> #include <linux/uaccess.h>
> #else
39a56,80
> #endif
> 
> /*
>  * Kernel 5.0 removed VERIFY_READ and VERIFY_WRITE and removed the first
>  * parameter of access_ok() which was set to VERIFY_READ or VERIFY_WRITE.
>  * That has been redundant since kernel 2.5.70, and even then it was only
>  * checked for kernels that support old 386 processors.
>  *
>  * Get rid of the first parameter and always pass VERIFY_WRITE for kernels
>  * prior to 5.0.  This will fail for old 386 processors on pre-2.5.70
>  * kernels if the memory region is not in fact writeable.
>  */
> #ifdef VERIFY_WRITE
> /* Pre 5.0 kernel. */
> static inline int _kcompat_access_ok(unsigned long addr, size_t size)
> {
>     /* Always use VERIFY_WRITE.  Most architectures ignore it. */
>     return access_ok(VERIFY_WRITE, addr, size);
> }
> /* Redefine access_ok() to remove first parameter. */
> #undef access_ok
> #define access_ok(addr, size) _kcompat_access_ok((unsigned long)(addr), (size))
> #endif
> 
> 
1922d1962
<                     VERIFY_WRITE,
1935d1974
<                     VERIFY_READ,
diff -r gsc_api/PlxSdk/Linux/Driver/Source.Plx9000/Dispatch.c gsc_api/PlxSdk/Linux/Driver/Source.Plx9000/Dispatch.c
41c41
< #include <asm/uaccess.h>
---
> #include <linux/uaccess.h>
diff -r gsc_api/PlxSdk/Linux/Driver/Source.Plx9000/SuppFunc.c gsc_api/PlxSdk/Linux/Driver/Source.Plx9000/SuppFunc.c
39c39
< #include <asm/uaccess.h>
---
> #include <linux/uaccess.h>
